### PR TITLE
node: api review adjustements part I

### PR DESCRIPTION
**Pull request by FloVanGH**

### CODE REVIEW

File: api/node/__test__/js_value_conversion.spec.ts [hunk 1]
```diff
@@ -237,6 +237,8 @@ test('get/set brush properties', (t) => {
     in-out property <brush> black: #000000;
     in-out property <brush> trans: transparent;
     in-out property <brush> ref: transparent;
+    in-out property <brush> linerar-gradient: @linear-gradient(90deg, #3f87a6 0%, #ebf8e1 50%, #f69d3c 100%);
```
Comment (tronical) [inline api/node/__test__/js_value_conversion.spec.ts:240]: `linerar` -> `linear` (here and in the test below)

File: api/node/index.ts [hunk 2]
```diff
@@ -4,40 +4,88 @@
...
+     * Returns the visibility state of the window. This function can return false even if you previously called show()
+     * on it, for example if the user minimized the window.
+     */
+    get isVisible(): boolean;
+
+    /** Returns the logical position of the window on the screen. */
+    get logicalPosition(): Point;
+
+    /** Sets the logical position of the window on the screen. */
+    set logicalPosition(position: Point);
```
Comment (tronical) [inline api/node/index.ts:50]: When there's a setter and a getter with the same type, wouldn't it be easier to write this as a property? So:

```
logicalPosition: Point;
```

File: api/node/index.ts [hunk 3]
```diff
@@ -4,40 +4,88 @@
...
+
+    /** Returns the physical size of the window on the screen, */
+    get physicalSize(): Size;
+
+    /** Sets the logical size of the window on the screen, */
+    set physicalSize(size: Size);
+}
+
+/**
+ * An image data type that can be displayed by the Image element.
```
Comment (tronical) [inline api/node/index.ts:72]: In the review we said "link to https://developer.mozilla.org/en-US/docs/Web/API/ImageData in the docs" :)

File: api/node/index.ts [hunk 4]
```diff
@@ -292,17 +333,67 @@ export class CompileError extends Error {
...
+ * ```js
+ * import * as slint from "slint-ui";
+ * let ui = slint.loadFile(".ui/main.slint");
+ * let main = new ui.Main();
+ * ```
  */
-export function loadFile(filePath: string): Object {
+export function loadFile(filePath: string, options?: LoadFileOptions) : Object {
+    // this is a workaround that fixes an issue there resources in slint files cannot be loaded if the
+    // file path is given as relative path
```
Comment (tronical) [inline api/node/index.ts:372]: This comment seems outdated?

File: api/node/src/interpreter/diagnostic.rs [hunk 5]
```diff
@@ -35,14 +35,14 @@ pub struct JsDiagnostic {
     /// Message for this diagnostic.
     pub message: String,
 
-    /// The line number in the .slint source file.
+    /// The line number in the .slint source file. The line number start with 1.
```
Comment (tronical) [inline api/node/src/interpreter/diagnostic.rs:38]: "start" -> "starts" here and below for the column.

File: api/node/src/types/brush.rs [hunk 6]
```diff
@@ -1,23 +1,39 @@
...
-use i_slint_core::{Brush, Color};
-use napi::bindgen_prelude::External;
+use i_slint_core::{graphics::GradientStop, Brush, Color};
+use napi::{bindgen_prelude::External, Error, Result};
 
 /// Color represents a color in the Slint run-time, represented using 8-bit channels for red, green, blue and the alpha (opacity).
-#[napi(js_name = Color)]
+#[napi(object, js_name = "Color")]
 pub struct JsColor {
+    /// Represents red channel of the color as u8 in the range 0..255.
```
Comment (tronical) [inline api/node/src/types/brush.rs:10]: "Represents red channel" -> "Represents the red channel" (here and below for the other four)

File: api/node/src/types/brush.rs [hunk 7]
```diff
@@ -217,11 +241,52 @@ impl JsBrush {
...
+    }
+}
 
-        println!("toString() is not yet implemented for gradient brushes.");
-        String::default()
+fn gradient_stops_to_string<'a>(stops: impl Iterator<Item = &'a GradientStop>) -> String {
+    let stops: Vec<String> = stops
+        .map(|s| {
+            format!(
+                "rgba({},{},{}, {}) {}%",
```
Comment (tronical) [inline api/node/src/types/brush.rs:270]: I suggest to have a space after each comma, not only the last one.

File: tests/cases/callbacks/handler_with_arg.slint [hunk 8]
```diff
@@ -84,11 +84,11 @@ try {
 assert.equal(instance.callback_emission_count, 0);
 
 
-/// also test setHandler
-instance.test_callback2.setHandler(function(a) {
+/// also test test_callback2
```
Comment (ogoffart) [inline tests/cases/callbacks/handler_with_arg.slint:87]:
```suggestion
/// also test assigning with a function
```
(as opposed to initializing in the constructor)

**Review by tronical — COMMENTED** [review]
Looks good overall. A few small nitpicks.

**Review by ogoffart — APPROVED** [review]

File: api/node/index.ts (hunk 2 above)
Comment (FloVanGH) [reply api/node/index.ts]: Yes make sense. Done.

File: api/node/__test__/js_value_conversion.spec.ts (hunk 1 above)
Comment (FloVanGH) [reply api/node/__test__/js_value_conversion.spec.ts]: done

File: api/node/index.ts (hunk 4 above)
Comment (FloVanGH) [reply api/node/index.ts]: removed

File: api/node/index.ts (hunk 3 above)
Comment (FloVanGH) [reply api/node/index.ts:60]: done.

File: api/node/src/interpreter/diagnostic.rs (hunk 5 above)
Comment (FloVanGH) [reply api/node/src/interpreter/diagnostic.rs]: done

File: api/node/src/types/brush.rs (hunk 6 above)
Comment (FloVanGH) [reply api/node/src/types/brush.rs]: done

File: api/node/src/types/brush.rs (hunk 7 above)
Comment (FloVanGH) [reply api/node/src/types/brush.rs]: done

**Review by tronical — APPROVED** [review]

**Pull request merged**